[jit] better message for bad type annotation

ppwwyyxx opened the pull request:
Summary:
```
ValueError: Unknown type annotation: 'typing.Sequence[torch.Tensor]' at  File "xxx.py", line 223
        images = [x["image"].to(self.device) for x in batched_inputs]
        images = [(x - self.pixel_mean) / self.pixel_std for x in images]
        images = ImageList.from_tensors(images, self.backbone.size_divisibility)
                 ~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
        return images
```

Otherwise have no clue where the error is.

Test Plan: sandcastle

Differential Revision: D24764886

ppwwyyxx commented:
ah the PR still has unrelated contents from a different PR.. but the diff is clean

SplitInfinity commented on torch/jit/annotations.py:
```diff
@@ -342,7 +342,7 @@ def ann_to_type(ann, loc):
     if the_type is not None:
         return the_type
-    raise ValueError(f"Unknown type annotation: '{ann}'")
+    raise ValueError(f"Unknown type annotation: '{ann}' at{loc.highlight()}")
```
```suggestion
    raise ValueError(f"Unknown type annotation: '{ann}' at {loc.highlight()}")
```

ppwwyyxx replied:
if you look at the output I got, there are already some spaces in `highlight()`. Is this change still needed?

SplitInfinity replied:
I'm not sure why there are already some spaces but it'll be good to have at least one in case there aren't spaces in the highlight.

SplitInfinity reviewed (commented):
It seems that this is stacked on top of #45645 - if you unstack I think we can land this separately.

ppwwyyxx committed:
[jit] better message for bad type annotation (#47464)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/47464

```
ValueError: Unknown type annotation: 'typing.Sequence[torch.Tensor]' at  File "xxx.py", line 223
        images = [x["image"].to(self.device) for x in batched_inputs]
        images = [(x - self.pixel_mean) / self.pixel_std for x in images]
        images = ImageList.from_tensors(images, self.backbone.size_divisibility)
                 ~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
        return images
```

Otherwise have no clue where the error is.

Test Plan: sandcastle

Differential Revision: D24764886

fbshipit-source-id: b28de389024c447b59eafd0b654d792fd3b15961
